src/codegen/expression.rs

```diff
@@ -1465,31 +1385,16 @@ fn abi_encode_call(
         vartab,
         opt,
     );
-    let args = args_iter
+    let mut args = args_iter
         .map(|v| expression(v, cfg, contract_no, func, ns, vartab, opt))
-        .collect();
-    let res = vartab.temp(
-        &pt::Identifier {
-            loc: *loc,
-            name: "encoded".to_owned(),
-        },
-        &Type::DynamicBytes,
-    );
-    tys.insert(0, Type::Bytes(4));
-    cfg.add(
-        vartab,
-        Instr::Set {
-            loc: *loc,
-            res,
-            expr: Expression::AbiEncode {
-                loc: *loc,
-                tys,
-                packed: vec![selector],
-                args,
-            },
-        },
-    );
-    Expression::Variable(*loc, Type::DynamicBytes, res)
+        .collect::<Vec<Expression>>();
+
+    let mut encoder_args: Vec<Expression> = Vec::with_capacity(args.len() + 1);
+    encoder_args.push(selector);
+    encoder_args.append(&mut args);
+    let mut encoder = create_encoder(ns, false);
+    let (encoded, _) = encoder.abi_encode(loc, encoder_args, ns, vartab, cfg);
+    encoded
```

Comment:
Seems like this code is just repeated 3x and could potentially be loaded off into a helper function or something?

Reply:
Good point